synapse/lib/layer.py

```diff
@@ -4581,113 +4581,209 @@ async def confirmLayerEditPerms(self, user, gateiden, delete=False):
+                    if not realform: # pragma: no cover
+                        mesg = f'Invalid form: {form}'
+                        raise s_exc.NoSuchForm(mesg=mesg, form=form)
 
-                        if delete:
-                            self.core.confirmPropDel(user, realprop, gateiden)
-                        else:
-                            self.core.confirmPropSet(user, realprop, gateiden)
+                    realprop = realform.prop(prop)
+                    if not realprop: # pragma: no cover
+                        mesg = f'Invalid prop: {form}:{prop}'
+                        raise s_exc.NoSuchProp(mesg=mesg, form=form, prop=prop)
 
-                    elif not prop and not allow_forms:
-                        user.confirm(perm_forms + (form,), gateiden=gateiden)
+                    if delete:
+                        self.core.confirmPropDel(user, realprop, gateiden)
+                    else:
+                        self.core.confirmPropSet(user, realprop, gateiden)
+
+                elif not allow_forms:
+                    user.confirm(perm_forms + (form,), gateiden=gateiden)
 
         # tagprops
         if not allow_tags:
-            async for byts, abrv in s_coro.pause(self.tagpropabrv.slab.scanByFull(db=self.tagpropabrv.name2abrv)):
-                info = s_msgpack.un(byts)
-                if None in info or len(info) != 3:
-                    continue
-
-                if self.layrslab.prefexists(abrv, db=self.bytagprop):
-                    perm = perm_tags + tuple(info[1].split('.'))
-                    user.confirm(perm, gateiden=gateiden)
+            async for tagprop in s_coro.pause(self.getTagProps()):
+                perm = perm_tags + tuple(tagprop[1].split('.'))
+                user.confirm(perm, gateiden=gateiden)
 
         # nodedata
         if not allow_ndata:
             async for abrv in s_coro.pause(self.dataslab.scanKeys(db=self.dataname, nodup=True)):
-                name, _ = self.getAbrvProp(abrv)
-                perm = perm_ndata + (name,)
+                if abrv[8:] == FLAG_TOMB:
+                    continue
+
+                key = self.core.getAbrvIndx(abrv[:8])
+                perm = perm_ndata + key
                 user.confirm(perm, gateiden=gateiden)
 
         # edges
         if not allow_edges:
-            async for verb in s_coro.pause(self.layrslab.scanKeys(db=self.byverb, nodup=True)):
-                perm = perm_edges + (verb.decode(),)
+            async for verb in s_coro.pause(self.getEdgeVerbs()):
+                perm = perm_edges + (verb,)
+                user.confirm(perm, gateiden=gateiden)
+
+        # tombstones
+        async for lkey in s_coro.pause(self.layrslab.scanKeysByPref(INDX_TOMB, db=self.indxdb, nodup=True)):
+            byts = self.core.indxabrv.abrvToByts(lkey[2:10])
+            tombtype = byts[:2]
+            tombinfo = s_msgpack.un(byts[2:])
+
+            if (tombtype, delete) == (INDX_PROP, True):
```

Comment:
seems like checking tombtype and then delete within that would be cleaner than checking it as a tuple like this